pkg/status/combinedstatus-resolution.go

```diff
@@ -827,3 +870,34 @@ func valueEqual(a, b *v1alpha1.Value) bool {
 		return false
 	}
 }
+
+func parseNumber(s string) (any, error) {
+	if intValue, err := strconv.ParseInt(s, 10, 64); err == nil {
+		return intValue, nil
+	}
+
+	if uintValue, err := strconv.ParseUint(s, 10, 64); err == nil {
+		return uintValue, nil
+	}
+
+	if floatValue, err := strconv.ParseFloat(s, 64); err == nil {
+		return floatValue, nil
+	}
+
+	return nil, fmt.Errorf("failed to parse number")
+}
+
+// numericEqual compares two numeric values. It is assumed that the types of a
+// and b are the same.
```

Comment:
I do not see why this assumption is justified. When the assumption is not true, this func will panic, right?

Reply:
Right, tried to save some type checks falsely by this line:
```
func valueEqual(a, b *v1alpha1.Value) bool {
	if a.Type != b.Type {
		return false
	}
```

Which is incorrect, thanks.